src/settings/global_user.rs

```diff
@@ -11,18 +12,37 @@ impl GlobalUser {
     pub fn new() -> Result<Self, failure::Error> {
         get_global_config()
     }
+    pub fn config_directory() -> Option<PathBuf> {
+        get_global_config_directory()
+    }
+}
+
+fn get_global_config_directory() -> Option<PathBuf> {
```

Comment:
I created this function because:
a) the logic for getting the config dir was duplicated in `commands/config.rs`, this way I can re-use it from one place.
b) it has some logic of it's own, doing the whole `exists` dance though in the future with a major version bump it will be a lot simpler I think returning an `Option` is nice so that the config command can print a pretty error.